### DISCUSSION TITLE
DevTools in Production

### DISCUSSION THREAD

**Discussion by TimoVink** [TanStack/query#1415] [Q&A] [answered]
Hi all,

Based on this line the docs:
> By default, React Query Devtools are not included in production bundles when `process.env.NODE_ENV === 'production'`, so you don't need to worry about excluding them during a production build.

I was expecting the devtools to automatically be disabled in Production builds. In practice using `create-react-app`  and `v3.2.0` of `react-query` with the following code:
``` jsx
<span>{process.env.NODE_ENV}</span>
<ReactQueryDevtools initialIsOpen={false} />
```

I'm seeing this:
![react-query](https://user-images.githubusercontent.com/694180/102131304-de958980-3e06-11eb-894c-23ee57910d9d.png)

Am I misunderstanding the documentation?

Thanks for any guidance, this is an extremely useful library!

**Comment by TkDodo** [comment, marked as answer]
Hmm I think we "lost" this feature when moving devtools from the dedicated repo to this repo. We had this code before:

```
if (process.env.NODE_ENV === 'production') {
  module.exports = {
    ReactQueryDevtools: function() {return null;},
    ReactQueryDevtoolsPanel: function() {return null;},
  }
} else {
  module.exports = require('./dist/react-query-devtools.development.js')
}
```

but I can't find this anywhere now. I think we need an issue, or even better, a PR to fix it :)

**Reply by TimoVink** [reply]
Thanks for the prompt response! Happy to take a look

**Reply by tannerlinsley** [reply]
Working on it now :)

**Reply by tannerlinsley** [reply]
Pretty sure I just fixed this with the latest commit/release. Let me know if it still happens

**Reply by Belco90** [reply]
Is this fix published? I don't see any new package yet after v3.2.0

**Reply by TkDodo** [reply]
this is the fix: https://github.com/tannerlinsley/react-query/commit/5e4ef153de642551d554047ce99124bfb58e254e

but it has not been released yet

**Reply by TimoVink** [reply]
I can confirm this seems to be fixed in the newly released `v3.3.0`. Thanks!
